asmcli/lib/context.sh

```diff
@@ -0,0 +1,83 @@
+#######
+# set_up_local_workspace does everything that the script needs to avoid
+# polluting the environment or current working directory
+#######
+set_up_local_workspace() {
+  local OUTPUT_DIR; OUTPUT_DIR="$(context_get-option-from-variable "${CONTEXT_JSON}" "OUTPUT_DIR")"
+  info "Setting up necessary files..."
+  if [[ -z "${OUTPUT_DIR}" ]]; then
+    info "Creating temp directory..."
+    OUTPUT_DIR="$(mktemp -d)"
+    if [[ -z "${OUTPUT_DIR}" ]]; then
+      fatal "Encountered error when running mktemp -d!"
+    fi
+    info ""
+    info "$(starline)"
+    info "No output folder was specified with --output_dir|-D, so configuration and"
+    info "binaries will be stored in the following directory."
+    info "${OUTPUT_DIR}"
+    info "$(starline)"
+    info ""
+    sleep 2
+  else
+    OUTPUT_DIR="$(apath -f "${OUTPUT_DIR}")"
+    if [[ ! -a "${OUTPUT_DIR}" ]]; then
+      if ! mkdir -p "${OUTPUT_DIR}"; then
+        fatal "Failed to create directory ${OUTPUT_DIR}"
+      fi
+    elif [[ ! -d "${OUTPUT_DIR}" ]]; then
+      fatal "${OUTPUT_DIR} exists and is not a directory, please specify another directory."
+    fi
+  fi
+  pushd "$OUTPUT_DIR" > /dev/null
+  CONTEXT_JSON="$(context_set-option-from-variable "${CONTEXT_JSON}" "OUTPUT_DIR" "${OUTPUT_DIR}")"
+  readonly OUTPUT_DIR
+
+  KUBECONFIG="asm_kubeconfig"
+  export KUBECONFIG
+
+  # Dump CONTEXT_JSON for the first time
+  # This file from now on will only be read and udpated.
+  echo "${CONTEXT_JSON}" >| CONTEXT.json
+}
+
+context_get-option-from-file() {
+  local FILE; FILE="${1}"
+  local OPTION; OPTION="${2}"
+
+  jq -r --arg FILE "${FILE}" --arg OPTION "${OPTION}" '.flags[$OPTION]' "${FILE}"
```

Comment:
I don't think the `--arg FILE "${FILE}"` is needed. (but good thinking on the `--arg`, it helps readability.)

Reply:
Done